frontend/public/components/resource-dropdown.tsx

```diff
@@ -22,9 +31,20 @@ const ResourceListDropdown_: React.SFC<ResourceListDropdownProps> = props => {
   const { selected, onChange, allModels, showAll, className, preferredVersions } = props;
 
   const items = allModels
-    .filter(model => {
+    .filter(({apiGroup, apiVersion, kind, verbs}) => {
+      // Remove blacklisted items.
+      if (blacklist.has(`${apiGroup}/${apiVersion}.${kind}`)) {
+        return false;
+      }
+
+      // Only show resources that can be listed.
+      if (!_.isEmpty(verbs) && !_.includes(verbs, 'list')) {
+        return false;
+      }
+
+      // Only show preferred version for resources in the same API group.
```

Comment:
Do you want to redeclare the functions in every iteration of filter?  Its prob not a big deal as its a set of a known size.

Reply:
other than that i think its an lgtm, though the tests are complaining 😦

Reply:
Good eye... It's a little difficult to pull out since it uses vars from the closure. Also not new to this PR.

Do you mind if I punt on that? The list should never be too large